src/main/java/org/spongepowered/common/entity/EntityUtil.java

```diff
@@ -696,11 +696,21 @@ public static Profession validateProfession(int professionId) {
                 // Vanilla behaviour - Delete the known bed location if invalid
                 bedPos = null; // null = remove location
             }
-            // Set the new bed location for the new dimension
-            int prevDim = playerIn.dimension; // Temporarily for setSpawnPoint
+
+            // First revision: Xakep_SDK, 28.01.18. MC Version: 1.12.2, API version: 7.1.0
+            // START REVISION
+            // This calls
+            // MixinEntityPlayer#onSetSpawnPoint(BlockPos pos, boolean forced, CallbackInfo ci);
+            // so bedPos will be set for target world.
+            // If we don't do this, MixinPlayerList#recreatePlayerEntity(EntityPlayerMP playerIn, int targetDimension, boolean conqueredEnd)
```

Comment:
If anything, you should be able to java doc link to the method instead of a comment link, sometimes these get lost in time.

Reply:
Fixed in next commit